client/src/components/QandA/QuestionsList/QuestionCard.jsx

```diff
@@ -1,32 +1,166 @@
+      this.setState({
+        answerCounter: Object.keys(this.props.answers).length
+      });
+    } else {
+      this.setState({
+        answerCounter: 2,
+        showMoreBtn: true
+      });
+    }
+  }
 
+  onSeeMoreClick() {
+    if (Object.keys(this.props.answers).length - this.state.answerCounter === 0) {
+      this.setState({
+        showCollapseBtn: true
+      });
+    } else if (Object.keys(this.props.answers).length - this.state.answerCounter <= 2) {
+      this.setState({
+        answerCounter: Object.keys(this.props.answers).length,
+        showCollapseBtn: true
+      });
+    } else {
+      this.setState({
+        answerCounter: this.state.answerCounter + 2
+      });
+    }
   }
 
-  return (
-    <div id='Question-Card'>
-      <QuestionDetails
-        asker={asker}
-        date={date}
-        helpful={helpful}
-        reported={reported}
-      />
-      <div className='symbol question-symbol'>Q:</div>
-      <QuestionBody body={body}/>
-      <div className='symbol answer-symbol'>A:</div>
-      <Answers answers={answers} key={ID * 3}/>
-      {Object.keys(answers).length > 2
-        ? <button onClick={onSeeMoreAnswersClick}>See more answers</button>
-        : <></>
-      }
-    </div>
-  );
+  onCollapseClick() {
+    // this.initialSetter();
+    if (Object.keys(this.props.answers).length <= 2) {
+      this.setState({
+        answerCounter: Object.keys(this.props.answers).length,
+        showCollapseBtn: false
+      });
+    } else {
+      this.setState({
+        answerCounter: 2,
+        showMoreBtn: true,
+        showCollapseBtn: false
+      });
+    }
+  }
+
+  render() {
+    return (
+      <div id='Question-Card'>
+        <QuestionDetails
+          asker={this.props.asker}
+          date={this.props.date}
+          helpful={this.props.helpful}
+          reported={this.props.reported}
+          question_id={this.props.question_id}
+          product_name={this.props.product_name}
+        />
+        <div className='symbol question-symbol'>Q:</div>
+        <QuestionBody body={this.props.body}/>
+        {Object.keys(this.props.answers).length > 0 ?
+          <>
+            <div className='symbol answer-symbol'>A:</div>
+            <Answers
+              answers={this.props.answers}
+              key={this.props.question_id * 3}
+              answerCounter={this.state.answerCounter}
+              showMore={this.state.showMoreBtn}
+              collapse={this.state.showCollapseBtn}
+            />
+            {
+              !this.state.showMoreBtn ? null
+                : !this.state.showCollapseBtn
+                  ? <button onClick={this.onSeeMoreClick}>See more answers</button>
+                  : <button onClick={this.onCollapseClick}>Collapse answers</button>
+            }
+          </>
+          : null
+        }
+      </div>
+    );
+  }
 }
 
+// const QuestionCard = ({ answers, asker, body, date, helpful, reported, ID }) => {
```

Comment:
are you saving this code for some use later?

Reply:
Nope. Will remove.